x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -880,19 +881,14 @@ static AuthorizedIndices resolveAuthorizedIndicesFromRole(
         }, name -> {
             final IndexAbstraction indexAbstraction = lookup.get(name);
             if (indexAbstraction == null) {
-                return false;
-            }
-            if (includeDataStreams) {
+                // missing but authorized resources should be handled downstream in the action handler, not here in the Security filter
+                return predicate.testMissingResource(name);
```

Comment:
"missing but authorized resources should be handled downstream..." ... so why is this security code testing for missing ?

Reply:
I've changed the code comment to:
```
// test access (by name) to a resource that does not currently exist
// the action handler must handle the case of accessing resources that do not exist
```
The original comment could only be understood in the context of the change, as a GH comment rather than in the code, apologies.